clang/tools/clang-linker-wrapper/ClangLinkerWrapper.cpp

```diff
@@ -554,6 +555,13 @@ Expected<StringRef> clang(ArrayRef<StringRef> InputFiles, const ArgList &Args) {
   if (Args.hasArg(OPT_embed_bitcode))
     CmdArgs.push_back("-Wl,--lto-emit-llvm");
 
+  if (HasSYCLOffloadKind) {
+    CmdArgs.push_back("--sycl-link");
```

Comment:
Some comments on what this option does (and the requirement of the associated `-Xlinker` options would help with understanding the need for the change.  It is a bit confusing as use of `-Xlinker` implies options being passed to the link step, but it is unclear what the link step is in this case.

Reply:
Added a comment in this commit: 6298a6a34c00e5f88b559732cbb3d2bac2fc6ce3

Thanks